src/Microsoft.AspNet.Razor/Common/HashCodeCombiner.cs

```diff
@@ -49,7 +49,8 @@ public HashCodeCombiner Add(object o)
 
         public HashCodeCombiner Add<TValue>(TValue value, IEqualityComparer<TValue> comparer)
         {
-            return Add(comparer.GetHashCode(value));
+            var hashCode = value != null ? comparer.GetHashCode(value) : 0;
+            return Add(0);
```

Comment:
Unfortunately the comparer performs a null check so this ends up throwing. I'll fix the hash code